C/services/south/south.cpp

```diff
@@ -940,6 +959,86 @@ void SouthService::configChange(const string& categoryName, const string& catego
 	}
 }
 
+/**
+ * Separate thread to run plugin_reconf, to avoid blocking 
+ * service's management interface due to long plugin_poll calls
+ */
+static void reconfThreadMain(void *arg)
+{
+	SouthService *ss = (SouthService *)arg;
+	Logger::getLogger()->info("reconfThreadMain(): Spawned new thread for plugin reconf");
+	while(1)
+	{
+		ss->handlePendingReconf();
+	}
+	Logger::getLogger()->info("reconfThreadMain(): plugin reconf thread exiting");
```

Comment:
And how does this ever happen?

Reply:
At south service shutdown only. Just indicates that this thread exited cleanly.

Reply:
It is not required to terminate, that's all this thread does. It just waits for new reconf operations. It waits for m_cvNewReconf condition variable inside handlePendingReconf() before it processes all pending reconf requests at that time.